python/gstaichi/lang/_fast_caching/args_hasher.py

```diff
@@ -90,7 +90,20 @@ def stringify_obj_type(path: tuple[str, ...], obj: object, arg_meta: ArgMetadata
         for k, v in obj.__dict__.items():
             _child_repr = stringify_obj_type((*path, k), v, ArgMetadata(Template, ""))
             if _child_repr is None:
-                print("not representable child", k, type(v), "path", path)
+                _logging.warn(
+                    f"""
+A kernel that has been marked as eligible for fast cache was passed 1 or more parameterss that are not,
+in fact, eligible for fast cache.
+
+Concretely, one of the parameters was a @ti.data_oriented objects, and one of its children was not eligible.
+
+The data oriented object was of type {type(obj)}.
+
+The child {k}={type(v)} was not eligible for usage with fast cache.
+
+Technical information: the path of the value was {path}.
+"""
+                )
```

Comment:
What's a lot of details now x) What about packing lines to avoid spreading on half of the terminal? 
```
A kernel that has been marked as eligible for fast cache was passed 1 or more parameters that are not, in fact, eligible for fast cache: one of the parameters was a @ti.data_oriented objects, and one of its children was not eligible.
The data oriented object was of type {type(obj)} and the child {k}={type(v)} was not eligible. For information, the path of the value was {path}.
```